src/main/java/com/ldtteam/structurize/util/BlockToItemMaps.java

```diff
@@ -0,0 +1,179 @@
+package com.ldtteam.structurize.util;
+
+import com.google.common.collect.ImmutableMap;
+import com.ldtteam.structurize.items.ModItems;
+import net.minecraft.block.AbstractFireBlock;
+import net.minecraft.block.AttachedStemBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.FlowerPotBlock;
+import net.minecraft.block.FlowingFluidBlock;
+import net.minecraft.item.Item;
+import net.minecraft.item.Items;
+import net.minecraft.util.IItemProvider;
+import net.minecraftforge.registries.ForgeRegistries;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Block to Item wrapper for blocks which should not be in vanilla blockToItem map
+ */
+public class BlockToItemMaps
+{
+    /**
+     * Blocks which are aquireable in different ways
+     */
+    private static final Map<Block, IItemProvider> SURVIVAL_HELPERS;
+
+    /**
+     * Creative map
+     */
+    private static final Map<Block, IItemProvider> CREATIVE_HELPERS;
+
+    /**
+     * Blocks which should be treated as something else during build requirements
+     */
+    private static final Map<Block, IItemProvider> BUILDING_HELPERS;
+
+    /**
+     * Blocks which should be treated as something else during build requirements
+     */
+    private static final Map<Block, IItemProvider> VANILLA_MAP;
+
+    static
+    {
+        final ImmutableMap.Builder<Block, IItemProvider> survival = ImmutableMap.builder();
+        final ImmutableMap.Builder<Block, IItemProvider> creative = ImmutableMap.builder();
+        final ImmutableMap.Builder<Block, IItemProvider> building = ImmutableMap.builder();
+
+        // TODO missing items: END_GATEWAY, END_PORTAL, NETHER_PORTAL
+        // MOVING_PISTON and PISTON_HEAD are airs everytime
+
+        // air blocks
+        register(creative, Blocks.CAVE_AIR, ModItems.vanillaCaveAir.get());
+        register(creative, Blocks.VOID_AIR, ModItems.vanillaVoidAir.get());
+
+        // frostwalker enchantment
+        register(building, Blocks.FROSTED_ICE, ((FlowingFluidBlock) Blocks.WATER).getFluid().getFilledBucket());
+
+        // missing growables
+        register(survival, Blocks.BAMBOO_SAPLING, Items.BAMBOO);
+        register(survival, Blocks.KELP_PLANT, Items.KELP);
+        register(survival, Blocks.TALL_SEAGRASS, Items.SEAGRASS);
+        register(survival, Blocks.TWISTING_VINES_PLANT, Items.TWISTING_VINES);
+        register(survival, Blocks.WEEPING_VINES_PLANT, Items.WEEPING_VINES);
+
+        // farmland and path can be dirt and grass block
+        register(building, Blocks.FARMLAND, Items.DIRT);
+        register(building, Blocks.GRASS_PATH, Items.DIRT);
+        register(building, Blocks.GRASS_BLOCK, Items.DIRT);
```

Comment:
mostly to Ray: here we should add builder overrides, so they are not affecting creative things etc.